core/quantity/src/Formatter/Formatter.ts

```diff
@@ -232,7 +234,7 @@ export class Formatter {
       }
     }
 
-    return compositeStrings.join(spec.format.spacer ? spec.format.spacer : "");
+    return compositeStrings.join((spec.format.spacer !== undefined) ? spec.format.spacer : " ");
```

Comment:
Hmm, maybe if a spacer is undefined, it would mean that it shouldn't be spaced? So, I see joining on an empty string like previously would make more sense

Reply:
formats have the spacer attribute set to `" "` anyways, so the only way its undefined would be if someone deliberately passed spacer as null to the format props, or if they explicitly want an empty string as their spacer

Reply:
good observation. Still, since spacer is allowed to be undefined, I guess there is no way to avoid this check. Or did you have a better idea? Previously, we always added a ":"  between components in addition to the spacer, which I removed. So before it was never really possible to have "no spacing". But now it can be done. I think if somebody really wants it, setting the spacer to "" is a good precondition.

Reply:
yes... now its possible to have no spacing we should properly support it. We can keep the check make the spacer set to "" when the check fails, and just gotta properly document that this fix enables no spacing if a user wants it (Then again, if no one ever raised the `:` issue in the past 3 years maybe no one needed it hah)